### PR TITLE
fix few bugs cause less accuracy of gguf format

### PR DESCRIPTION
**Pull request by n1ck-guo**
## Description

Please briefly describe your main changes, the motivation.

## Type of Change

Bug fix

## Related Issues

Fixes or relates to #

## Checklist Before Submitting

- [ ] My code has been tested locally.
- [ ] Documentation has been updated as needed.
- [ ] New or updated tests are included where applicable.
- [ ] The CUDA CI has passed. You can trigger it by commenting `/azp run Unit-Test-CUDA-AutoRound`.

--options "gguf:q2_k_s,gguf:q4_k_s"  --target_bits 3.5 --ignore_scale_zp_bits
|   |gpqa_diamond|math_500|
|---|---|---|
|qwen3-8b f16| 0.4394 | 0.786 |
|gguf format| 0.4545|0.77|

### CODE REVIEW

File: auto_round/compressors/base.py [hunk 1]
```diff
@@ -641,28 +637,25 @@ def configure_layer_config(self, enable_gguf_official_mixed: bool | None = True)
...
-                ]
-            predefined_ignore_layers = compress_layer_names(predefined_ignore_layers)
-            if predefined_ignore_layers:
-                logger.info(f"Using predefined ignore_layers: {compressed_predefined_ignore_layers}")
-                tmp_str = predefined_ignore_layers.replace(" ", "")
-                if self.ignore_layers == "":
-                    self.ignore_layers = tmp_str
-                else:
-                    self.ignore_layers += "," + tmp_str
+        predefined_ignore_layers = get_predefined_ignore_layers(self.model_context.model)
```
Comment (wenhuach21) [inline auto_round/compressors/base.py]: Why change this? The predefined ignore layers are mainly intended for vLLM, so I don't think they should be applied to the GGUF format. The original code already handled this correctly.

File: auto_round/data_type/gguf.py [hunk 2]
```diff
@@ -461,21 +461,19 @@ def search_gguf_scale_min_asym(tensor, bits=4, scale_dtype=torch.float16, imatri
...
-
-        weights = weights.expand(tensor.numel() // weights.numel(), -1)
-        quant_weights = weights.reshape(tensor.shape)
-
+        weights = weights.expand(tensor.numel() // weights.numel(), -1).reshape(tensor.shape)
+
+        tensor_blocks = tensor.reshape(-1, groups_per_block, group_size)
+        weight_blocks = weights.reshape_as(tensor_blocks)
+        sigma_factor = 1 if bits == 2 else 2
+        sigma2 = sigma_factor * torch.sum(torch.pow(tensor_blocks, 2), dim=(-1, -2), keepdim=True) / QK_K
```
Comment (wenhuach21) [inline auto_round/data_type/gguf.py]: Please don't change any algorithm unless we have sufficient data demonstrating a clear advantage. This code was previously removed, which implies our preliminary tests did not show any measurable gains.

File: auto_round/compressors/base.py (hunk 1 above)
Comment (n1ck-guo) [reply auto_round/compressors/base.py]: revert

File: auto_round/data_type/gguf.py (hunk 2 above)
Comment (n1ck-guo) [reply auto_round/data_type/gguf.py]: After verification, the original one proved to be more accurate, and the relevant modifications have been revert.

File: auto_round/formats.py [hunk 3]
```diff
@@ -95,12 +95,23 @@ def _check_compatibility(formats: list[str], ar: BaseCompressor):
...
-            tmp_format_name = gguf_format_name.lower() if "fake" not in formats else f"{gguf_format_name.lower()},fake"
-            logger.warning(
-                f"reset format {','.join(formats)} to {tmp_format_name} "
-                f"since scheme {gguf_format_name} can only be exported to format {gguf_format_name.lower()} or fake"
-            )
-            formats = tmp_format_name.split(",")
+            has_gguf_format = any(f.lower().startswith("gguf") for f in formats if f.lower() != "fake")
+            if has_gguf_format:
+                logger.warning(
+                    f"scheme {gguf_format_name} is GGUF, but format {','.join(formats)} specifies "
```
Comment (wenhuach21) [inline auto_round/formats.py:101]: how about resetting format to match scheme

File: auto_round/data_type/gguf.py [hunk 4]
```diff
@@ -462,20 +462,10 @@ def search_gguf_scale_min_asym(tensor, bits=4, scale_dtype=torch.float16, imatri
...
-
         weights = weights.expand(tensor.numel() // weights.numel(), -1)
         quant_weights = weights.reshape(tensor.shape)
-
         quant_weights = _imatrix_handle_zero(quant_weights, tensor, bits)
 
-        # sigma2 = torch.sum(torch.pow(tensor, 2), dim=-1, keepdim=True) / QK_K
-        # if imatrix is None:
-        #     av_x = torch.sqrt(sigma2)
-        #     quant_weights = torch.abs(av_x + tensor * tensor)
```
Comment (wenhuach21) [inline auto_round/data_type/gguf.py:474]: better keep it for debugging algorithm in the future

**Review by wenhuach21 — APPROVED** [review]

File: auto_round/algorithms/quantization/base.py [hunk 5]
```diff
@@ -56,61 +56,52 @@ def quantize_layer_via_rtn(self, layer_name: str, disable_opt_rtn: bool | None =
...
-        if (
-            self.compress_context.is_immediate_packing
-            and self.compress_context.formats[0].is_gguf()
-            and not getattr(self.config, "disable_opt_rtn", False)
-        ):
+        try:
+            if disable_opt_rtn is None:
+                disable_opt_rtn = bool(getattr(self.config, "disable_opt_rtn", False))
+            if (
+                not disable_opt_rtn
```
Comment (wenhuach21) [inline auto_round/algorithms/quantization/base.py:63]: The try block wraps too much code. Could you narrow its scope so that it only covers the statements that may actually throw an exception?

**Review by wenhuach21 — APPROVED** [review]
